src/components/GuidedSetup/index.js

```diff
@@ -1,47 +1,100 @@
-import { GuidedStepEnum } from '../../utils/enums';
 
-function GuidedSetup() {
-  const guidedStep = useSelector(userSelectors.selectGuidedStep);
-  const dispatch = useDispatch();
+const useIsUnsupportedBrowser = () => {
+  const [isBrave, setIsBrave] = useState(null);
 
   useEffect(() => {
-    dispatch(authActions.guidedStep.call());
-  }, [dispatch]);
+    (async () => {
+      setIsBrave(!!(navigator.brave && await navigator.brave.isBrave()));
+    })();
+  }, [setIsBrave]);
 
+  return isBrave;
+};
+
+function GuidedSetupWrapper({ children }) {
   return (
     <div className={styles.guidedSetupWrapper}>
       <div className={styles.componentWrapper}>
-        {(!guidedStep || guidedStep?.component === GuidedStepEnum.LOADING) && <LoadingComponent />}
-        {guidedStep?.component === GuidedStepEnum.UNSUPPORTED_BROWSER
-         && (
-         <UnsupportedBrowserNoticeComponent />
-         )}
-        {guidedStep?.component === GuidedStepEnum.NO_WALLETS_AVAILABLE && <NoWalletsDetectedInBrowser />}
-        {guidedStep?.component === GuidedStepEnum.NO_CONNECTED_WALLET
-         && <NoConnectedWalletComponent walletList={guidedStep?.data?.wallets} userId={guidedStep?.data?.userId} />}
-        {guidedStep?.component === GuidedStepEnum.MISSING_WALLET
-         && (
-         <MissingWalletComponent
-           walletList={guidedStep?.data?.wallets}
-           registeredAddress={guidedStep?.data?.centralizedWalletAddress}
-           userId={guidedStep?.data?.userId}
-         />
-         )}
+        {children}
       </div>
     </div>
   );
 }
 
+function GuidedSetup({ children }) {
+  const [acceptedBrowser, setAcceptedBrowser] = useState(localStorage.getItem('unsupportedBrowserAcceptedByUser'));
+  const isSessionReady = useSelector(userSelectors.selectIsSessionReady);
+  const extensions = useSelector(blockchainSelectors.extensionsSelector);
+  const wallets = useSelector(blockchainSelectors.allWalletsSelector);
+  const userWalletAddress = useSelector(userSelectors.selectWalletAddress);
+  const userId = useSelector(userSelectors.selectUserId);
+  const isUnsupportedBrowser = useIsUnsupportedBrowser();
+
+  const isLoading = !isSessionReady
+    || extensions === null
+    || wallets === null
+    || isUnsupportedBrowser === null;
+
+  const onUnsupportedBrowserAccept = () => {
+    localStorage.setItem('unsupportedBrowserAcceptedByUser', true);
+    setAcceptedBrowser(true);
+  };
+
+  if (isLoading) return <LoadingComponent />;
+
+  if (!acceptedBrowser && isUnsupportedBrowser) {
+    return (
+      <GuidedSetupWrapper>
+        <UnsupportedBrowserNoticeComponent onAccept={onUnsupportedBrowserAccept} />
+      </GuidedSetupWrapper>
+    );
+  }
+
+  if (!userId) return children;
+
+  if (extensions.length === 0) {
+    return (
+      <GuidedSetupWrapper>
+        <NoWalletsDetectedInBrowser />
+      </GuidedSetupWrapper>
+    );
+  }
+
+  if (wallets.length === 0) {
```

Comment:
Opps, I think I misunderstood this one. This should be checking if `userWalletAddress` is empty.

Reply:
This should be with extensions check, if there is no extension, or any wallets, it's should show NoWalletDetectedInBrowser component.
But you are right, if userWalletAdress is empty it's should show NoConnectedWalletComponent.